hydra-node/src/Hydra/Network.hs

```diff
@@ -39,13 +42,23 @@ import Text.Show (Show (show))
 -- * Hydra network interface
 
 -- | Handle to interface with the hydra network and send messages "off chain".
-newtype Network m msg = Network
+data Network m msg = Network
   { -- | Send a `msg` to the whole hydra network.
     broadcast :: msg -> m ()
+  , modifyPeers :: forall a. (Set Host -> (a, Set Host)) -> m a
   }
 
+-- | Returns the list of connected peers in the hydra network.
+getPeers :: Network m msg -> m (Set Host)
+getPeers n = modifyPeers n (\ps -> (ps, ps))
+
+-- | Modify the list of peers in the hydra network,
+-- causing the network to reinitialize and reconnect to the new list of peers.
+setPeers :: Network m msg -> Set Host -> m ()
+setPeers n ps = modifyPeers n (const ((), ps))
+
 instance Contravariant (Network m) where
-  contramap f (Network bcast) = Network $ \msg -> bcast (f msg)
+  contramap f network@Network{broadcast} = network{broadcast = broadcast . f}
```

Comment:
Do we use this type class somewhere? In a way, Network became only partially contravariant now and we should think about not declaring as such, but to the decoration explicitly (as done in withHeartbeat)

Reply:
Logging. Contra-tracers makes use of this.

Reply:
We do `contramap` on the traces, yes. To be more precise: Are we using this instance somewhere? i.e. the `contramap` on `Network m` components.